torch/csrc/autograd/custom_function.cpp

```diff
@@ -26,6 +29,149 @@ Variable VariableInfo::zeros(at::OptionalDeviceGuard& device_guard) const {
   }
 }
 
+void _foreach_tensor(
+    const std::function<void (size_t, size_t, const at::Tensor&)>& fn,
+    torch::jit::Stack* stack,
+    size_t stack_start,
+    size_t size) {
+  // Enumerate over tensors in a stack, including ones in TensorLists
+  int idx = 0;
+  for (const auto i : c10::irange(size)) {
+    auto& ivalue = (*stack)[stack_start + i];
+    if (ivalue.isTensor()) {  // true for optional tensor that has value
+      const auto& tensor = ivalue.toTensor();
+      fn(idx, i, tensor);
+      idx++;
+    } if (ivalue.isTensorList()) {
+      for (const auto& tensor : ivalue.toTensorVector()) {
+        fn(idx, i, tensor);
+        idx++;
+      }
+    }
+  }
+}
+
+void autogradNotImplementedFallback(const c10::OperatorHandle& op, DispatchKeySet dispatch_keys, torch::jit::Stack* stack) {
```

Comment:
no action needed, but this would be a really good use for https://github.com/pytorch/pytorch/issues/45203